snmp/datadog_checks/snmp/snmp.py

```diff
@@ -207,44 +210,41 @@ def fetch_results(self, config, all_oids, bulk_oids):
         results.default_factory = None
         return results, error
 
-    def fetch_oids(self, config, oids, enforce_constraints):
-        # UPDATE: We used to perform only a snmpgetnext command to fetch metric values.
-        # It returns the wrong value when the OID passeed is referring to a specific leaf.
-        # For example:
-        # snmpgetnext -v2c -c public localhost:11111 1.3.6.1.2.1.25.4.2.1.7.222
-        # iso.3.6.1.2.1.25.4.2.1.7.224 = INTEGER: 2
-        # SOLUTION: perform a snmpget command and fallback with snmpgetnext if not found
-        error = None
+    def fetch_oids(
+        self,
+        config,  # type: InstanceConfig
+        oids,  # type: typing.Sequence[ObjectType]
+        enforce_constraints,  # type: bool
+    ):
+        # type: (...) -> typing.Tuple[typing.List[ObjectType], typing.Optional[str]]
+        error = None  # type: typing.Optional[str]
         first_oid = 0
-        all_binds = []
+        all_binds = []  # type: typing.List[ObjectType]
+
         while first_oid < len(oids):
             try:
+                # SNMP hosts can respond to requests for a batch of OIDs, so we
+                # use that to make less round trips to the host.
                 oids_batch = oids[first_oid : first_oid + self.oid_batch_size]
-                self.log.debug('Running SNMP command get on OIDS %s', oids_batch)
-                error_indication, error_status, _, var_binds = next(
```

Comment:
I think you're losing the `error_status` here, you don't extract it from the result.

Reply:
We were not using it initially though, were we? In any case, it's not lost — we can still access it via `result.error_status`.